src/Neogma.ts

```diff
@@ -75,4 +83,44 @@ export class Neogma {
   ): Promise<T> => {
     return getRunnable<T>(runInExisting, callback, this.driver);
   };
+
+  public generateModelFromMetadata = (
+    metadata: NeogmaModelMetadata,
+  ): NeogmaModel<any, any, any, any> => {
+    const parsedMetadata = parseModelMetadata(metadata);
+    const relations = parsedMetadata.relationships;
+    if (relations) {
+      for (const relation in relations) {
+        const relatedModel = metadata.relations[relation].model;
+        if (relatedModel !== 'self') {
+          const relatedModelLabel = relatedModel['name'];
+          if (this.modelsByName[relatedModelLabel]) {
+            relations[relation].model = this.modelsByName[relatedModelLabel];
+          } else {
+            const relatedModelMetadata = getRelatedModelMetadata(relatedModel);
+            relations[relation].model =
+              this.generateModelFromMetadata(relatedModelMetadata);
+          }
+        }
+      }
+    }
+
+    return ModelFactory(parsedMetadata, this) as unknown as NeogmaModel<
+      any,
+      any,
+      any,
+      any
+    >;
```

Comment:
I think that this casting can be removed?
```suggestion
    return ModelFactory(parsedMetadata, this)
```

Reply:
I've updated that recently actually to help with the issue where types weren't working on the created models, you'll see in the add model function, I added a generic to allow for proper generation of the model class props type. It's actually still WIP because I'm still trying to figure out a good way to impl the relations

Reply:
Sounds good 👌